akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -1194,10 +1215,14 @@ class DDataShardCoordinator(
   // which was scheduled by previous watchStateActors
   def waitingForStateInitialized: Receive = {
     case StateInitialized =>
-      unstashGetShardHomeRequests()
-      unstashAll()
-      stateInitialized()
-      activate()
+      log.debug("StateInitialized") // FIXME remove
+      update(ShardCoordinatorInitialized) { _ =>
+        log.debug("ShardCoordinatorInitialized completed") // FIXME remove
+        unstashGetShardHomeRequests()
+        unstashAll()
+        stateInitialized()
+        activate()
+      }
```

Comment:
this is for the suggestion for scenario 3: https://github.com/akka/akka/issues/28856#issuecomment-607750168